File: 20-02-2024/aproximacao-pi/pi.c
#include <stdio.h>
#include <math.h>

double calc(int length, int index, double agg) {
    if (length == index) {
        return agg;
    }

    int value = (index * 2) + 1;
    double seq = 1 / pow(value, 3);

    if (index == 0) {
        return calc(length, (index + 1), seq);
    }

    if (index % 2 == 0) {
        return calc(length, (index + 1), agg + seq);
    }

    return calc(length, (index + 1), agg - seq);
}

int main() {
    int n;

    printf("Informe o número de termos: ");
    scanf("%d", &n);

    double sequence = calc(n, 0, 0);
    double pi = cbrt(sequence * 32); // Elevar a 1/3 e pegar a raiz cúbica do número é a mesma coisa.

    printf("\n%.5f\n", pi);

    return 0;
}

File: 20-02-2024/aproximacao-pi/Pi.java
import java.util.Scanner;

public class Pi {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.println("Informe o número de termos: ");
            final int lenght = scanner.nextInt();

            System.out.println(String.format("%.5f", getPI(lenght)));
        }
    }

    public static double calculate(int length) {
        double agg = 0;

        for (int i = 0; i < length; i++) {
            final int value = (i * 2) + 1;
            final double seq = 1 / Math.pow(value, 3);

            if (i == 0) {
                agg += seq;

                continue;
            }
 
            if (i % 2 == 0) {
                agg -= seq;
            } else {
                agg += seq;
            }

        }

        return agg;
    }

    public static double getPI(int length) {
        return Math.cbrt(32 * calculate(length));
    }

}